### PR TITLE
WFLY-3384: fixing group caches jndi names, were children of the related ...

### PR DESCRIPTION
**Pull request by emmartins**
...non-context group entry

### CODE REVIEW

**Comment by emmartins** [comment]
@pferraro needs to review if this change as any chance to break stuff, for instance due to changed names being used in config files

File: clustering/spi/src/main/java/org/wildfly/clustering/spi/GroupServiceName.java [hunk 1]
```diff
@@ -40,4 +40,15 @@ public String toString() {
             return "group";
         }
     },
+    GROUP_CACHE() {
```
Comment (pferraro) [inline clustering/spi/src/main/java/org/wildfly/clustering/spi/GroupServiceName.java]: What is this for?  This change shouldn't have anything to do with the clustering api.

File: clustering/spi/src/main/java/org/wildfly/clustering/spi/CacheGroupServiceName.java [hunk 2]
```diff
@@ -34,10 +34,10 @@ public ServiceName getServiceName(String container, String cache) {
             return GroupServiceName.NODE_FACTORY.getServiceName(container).append(cache);
         }
     },
-    GROUP() {
+    GROUP_CACHE() {
```
Comment (pferraro) [inline clustering/spi/src/main/java/org/wildfly/clustering/spi/CacheGroupServiceName.java]: This name makes no sense - this value provides the service name of a org.wildfly.clustering.group.Group object.  This is not a cache.

**Comment by pferraro** [comment]
I don't understand why you needed to change the ServiceNameFactory enums.  The context names for these services is not directly related to their ServiceNames (even though they resemble each other), and can therefore be changed independently.

**Comment by pferraro** [comment]
OK - I actually read through the jira this time, and now I understand the problem you're trying to solve... :)
In summary, the clustering API exposes 2 different "groups":
1. A channel "group", corresponding to a channel view
2. An cache "group", corresponding to a cache view (which is always a subset of the channel view)

The 1st kind of group exists per channel and therefore per cache container (i.e we install a fork channel per cache container), whereas the 2nd kind of group exists per cache per cache container.  Consequently, there can exists a ServiceName "jboss.clustering.group.ejb" as well as a ServiceName "jboss.clustering.group.ejb.dist".  According to the current jndi naming logic, these correspond to a "java:jboss/clustering/group/ejb" binding as well as a "java:jboss/clustering/group/ejb/dist" binding - which cannot co-exist.

Rather than muck about with the names, there is a better solution, I think.  The "ejb" group represents a fork channel of the "ee" group.  These are actually equivalent.  So, rather than create a new service for the "ejb" group, this instead be installed as an alias of the "ee" group.  This way there is never a name/parent collision.

@emmartins You can assign this issue to me - and I'll make the above change.

**Comment by pferraro** [comment]
Actually, the "ejb" group is already an alias of the "ee" group. The issue is that there should only be an alias for the group service, not the jndi binding.

**Pull request closed without merging**